contrib/intel/jenkins/build.py

```diff
@@ -69,6 +69,30 @@ def build_fabtests(libfab_install_path, mode):
     common.run_command(['make', '-j32'])
     common.run_command(['make', 'install'])
 
+def extract_mpich(mpitype):
+
+    if (mpitype == 'mpich'):
+        src_dir = 'mpich'
+        dest = 'mpich_temp'
+        mpich_tar = 'mpich-4.1.tar.gz'
+    elif (mpitype == 'impi'):
+        src_dir = 'impi_mpichtest'
+        dest = 'impi_mpichtest'
+        mpich_tar = 'mpich-test.tar.gz'
+    else:
+        print(f"Invalid mpi type {mpitype}")
+        exit
+
+    cwd = os.getcwd()
+    if (os.path.exists(f'{cloudbees_config.build_dir}/{dest}') == True):
+        common.run_command(['rm','-rf', f'{cloudbees_config.build_dir}/{dest}'])
+    os.makedirs(f'{cloudbees_config.build_dir}/{dest}')
+    shutil.copy(f'{cloudbees_config.scm_dir}/{src_dir}/{mpich_tar}',
+                    f'{cloudbees_config.build_dir}/{dest}/')
+    os.chdir(f'{cloudbees_config.build_dir}/{dest}/')
+    common.run_command(['tar','-xvf', f'{cloudbees_config.build_dir}/{dest}/{mpich_tar}'])
```

Comment:
this should probably be a named tar extraction so when mpich version upgrades in the future it doesn't change the name of the extracted tarfile that tests.py expects

Reply:
what is "this"? :)

Reply:
line 93 should probably extract the tar to a common name

Reply:
names  are made generic now.